core/src/test/scala/org/apache/spark/status/AppStatusListenerSuite.scala

```diff
@@ -1860,15 +1860,16 @@ abstract class AppStatusListenerSuite extends SparkFunSuite with BeforeAndAfter
 
   private def newAttempt(orig: TaskInfo, nextId: Long): TaskInfo = {
     // Task reattempts have a different ID, but the same index as the original.
-    new TaskInfo(nextId, orig.index, orig.attemptNumber + 1, time, orig.executorId,
+    new TaskInfo(
+      nextId, orig.index, orig.attemptNumber + 1, orig.partitionId, time, orig.executorId,
       s"${orig.executorId}.example.com", TaskLocality.PROCESS_LOCAL, orig.speculative)
   }
 
   private def createTasks(count: Int, execs: Array[String]): Seq[TaskInfo] = {
     (1 to count).map { id =>
       val exec = execs(id.toInt % execs.length)
       val taskId = nextTaskId()
-      new TaskInfo(taskId, taskId.toInt, 1, time, exec, s"$exec.example.com",
+      new TaskInfo(taskId, taskId.toInt, 1, taskId.toInt, time, exec, s"$exec.example.com",
         TaskLocality.PROCESS_LOCAL, id % 2 == 0)
```

Comment:
Looks like this method is incorrect to begin with.
task index should be < number of tasks in stage and partition id should be < number of partitions.

We should change both to `id` instead.
Thoughts @stczwd, @Ngone51, @cloud-fan ?

Reply:
[here](https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/status/AppStatusListenerSuite.scala#L1127) use createTasks to genenrate different tasks. If we change `taskId` to `id`, this should be changed.